runtime/src/plugin.rs

```diff
@@ -259,7 +259,10 @@ impl Plugin {
         if let Some(init) = self.get_func("hs_init") {
             if let Some(cleanup) = self.get_func("hs_exit") {
                 if init.typed::<(i32, i32), (), _>(&self.memory.store).is_err() {
-                    trace!("hs_init found, but has the incorrect type");
+                    trace!(
+                        "hs_init found with type {:?}, which doesn't match the type exported by the Haskell runtime", 
```

Comment:
I'll push up another message here -- sorry to nit. My hesitation is that this could mislead someone who has no reason to think about the Haskell runtime. 

I know it would be unlikely for a non-Haskell module to export both `hs_init` and `hs_exit`, and therefore meet BOTH these conditions... but in the case that it did happen, we would be adding unrelated information to a log.

IMO, we don't need to say anything about the Haskell runtime until after `return Some(Runtime::Haskell { init, cleanup });`, where we can be pretty sure that whomever is looking at the `trace` message should see something about Haskell.